Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -4193,6 +4193,17 @@ public boolean moveCardToExileWithInfo(Card card, UUID exileId, String exileName
         return result;
     }
 
+    @Override
+    public Cards millCards(int toMill, Ability source, Game game) {
+        GameEvent event = GameEvent.getEvent(EventType.MILL_CARDS, getId(), source.getSourceId(), getId(), toMill);
+        if (game.replaceEvent(event)) {
+            return new CardsImpl();
+        }
+        Cards cards = new CardsImpl(this.getLibrary().getTopCards(game, event.getAmount()));
+        this.moveCards(cards, Zone.GRAVEYARD, source, game);
+        return cards;
```

Comment:
Returns all cards, but old code return only successfully moved (`moveCards(cards, Zone.GRAVEYARD`). Also some mill's results now have additional filters like `.filter(card -> game.getState().getZone(card.getId()) == Zone.GRAVEYARD)`. 

Maybe it's a redundant code and it can be replaced to old one (returns successfully moved and no filter)?

Reply:
The problem is that it can depend on how things get moved, and as far as I know the cards will at the very least leave the library, so it's best to just return all of them.